compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -323,6 +323,9 @@ TR::Instruction *loadConstant(TR::CodeGenerator *cg, TR::Node * node, int64_t va
       return loadConstant(cg, node, (int32_t)value, trgReg, cursor, isPicSite);
       }
 
+   if (useTOC && cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
```

Comment:
in theory, you don't need to check useTOC.  i.e.  loadFromConstantSnippet should be good for both scenarios of useTOC being true or false.  the constantSnippet is accommodating isPicSite being true or false anyway.  If there is runtime compensation needed, the compensation will be done on the constant, instead of the instruction itself. Maybe take this improvement as a low priority item.

Reply:
My understanding of this was that this existed since some pieces of code would assume that the generated code would look a certain way when `useTOC` was set to false. However, upon further investigation, I figured out that the only time that `useTOC` was false was when loading the address of the pTOC itself under system linkage on PPC64LE. Since that's obviously not relevant to P10, I've gone ahead and removed this condition.